pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -511,13 +511,21 @@ func (r *MemberRollReconciler) reconcileNamespaces(ctx context.Context, namespac
 			if namespacesToReconcile.Has(ns) {
 				err = reconciler.reconcileNamespaceInMesh(ctx, ns)
 				if err != nil {
-					if errors.IsNotFound(err) || errors.IsGone(err) { // TODO: this check should be performed inside reconcileNamespaceInMesh
+					if errors.IsNotFound(err) || errors.IsGone(err) {
 						reqLogger.Info("namespace to configure with mesh is missing", "namespace", ns)
 						r.EventRecorder.Event(
 							smmr,
 							corev1.EventTypeWarning,
 							string(eventReasonNamespaceMissing),
 							fmt.Sprintf("Namespace to configure with mesh is missing: %s", ns))
+					} else if IsTerminating(err) {
+						// We don't treat terminating namespaces as errors, because they are eventually deleted and we'll
+						// then log the NotFound error. If we were to log terminating namespaces as errors, we'd
+						// end up logging a bunch of transient errors that no-one cares about. This typically happens
+						// when someone deletes a member namespace that contains a ServiceMeshMember. The namespace
+						// is immediately marked as Terminating, but the servicemeshmember controller doesn't immediately
+						// remove it from SMMR.spec.members. The servicemeshmemberroll controller would then try to
+						// reconcile the namespace and log an error.
```

Comment:
Do we still need to remove our finalizer on the ServiceMeshMember?

Reply:
Yes. But that's done by the smm controller.